examples/seismic/model.py

```diff
@@ -52,53 +53,96 @@ def initialize_damp(damp, nbl, spacing, abc_type="damp"):
 
 class PhysicalDomain(SubDomain):
 
-    name = 'phydomain'
+    name = 'physdomain'
 
-    def __init__(self, nbl):
+    def __init__(self, so, fs=False):
         super(PhysicalDomain, self).__init__()
-        self.nbl = nbl
+        self.so = so
+        self.fs = fs
+
+    def define(self, dimensions):
+        map_d = {d: d for d in dimensions}
+        if self.fs:
+            map_d[dimensions[-1]] = ('middle', self.so, 0)
+        return map_d
+
+
+class FSDomain(SubDomain):
+
+    name = 'fsdomain'
+
+    def __init__(self, so):
+        super(FSDomain, self).__init__()
+        self.size = so
 
     def define(self, dimensions):
-        return {d: ('middle', self.nbl, self.nbl) for d in dimensions}
+        """
+        Definition of the top part of the domain for wrapped indices FS
+        """
+        z = dimensions[-1]
+        map_d = {d: d for d in dimensions}
+        map_d.update({z: ('left', self.size)})
+        return map_d
 
 
 class GenericModel(object):
     """
     General model class with common properties
     """
     def __init__(self, origin, spacing, shape, space_order, nbl=20,
-                 dtype=np.float32, subdomains=(), bcs="damp", grid=None):
+                 dtype=np.float32, subdomains=(), bcs="damp", grid=None,
+                 fs=False):
         self.shape = shape
         self.space_order = space_order
         self.nbl = int(nbl)
         self.origin = tuple([dtype(o) for o in origin])
+        self.fs = fs
+        if self.fs:
+            warning("Freesurface is only supported for isotropic acoustic solver")
```

Comment:
Should we not:
- Not have a warning if using iso-acoustic
- Throw a not implemented error if not using isco-acoustic
?

Reply:
I can try, the model doesn't really know which physics it is at this point so would need to move it to each example. I can just remove it as would just be `False` for all other examples